sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/rules/RuleExecutor.scala

```diff
@@ -56,12 +55,16 @@ abstract class RuleExecutor[TreeType <: TreeNode[_]] extends Logging {
   case class FixedPoint(maxIterations: Int) extends Strategy
 
   /** A batch of rules. */
-  protected case class Batch(name: String, strategy: Strategy, rules: Rule[TreeType]*)
+  case class Batch[TreeType <: TreeNode[_]]
```

Comment:
I'd probably just make these top level classes if we are going to pull them out.  There is not a lot of stuff in the `rules` package.